server/server_processor.cpp

```diff
@@ -104,12 +118,78 @@ void nibaserver::server_processor::process(nibashared::message_createchar &req)
         BOOST_LOG_SEV(logger_, sev::info)
             << "User " << *(session_.userid) << " created an character " << req.player;
         // grab magic&equips, might not be the best way but anyway
-        std::tie(session_.magics, session_.equips) = db_.get_aux(req.player.name, yield_);
+        std::tie(session_.magics, session_.equips, session_.equipped_magic_ids) =
+            db_.get_aux(req.player.name, yield_);
         req.magics = session_.magics;
         req.equips = session_.equips;
+        req.equipped_magic_ids = session_.equipped_magic_ids;
+        BOOST_LOG_SEV(logger_, sev::info)
+            << "User " << *(session_.userid) << " magics " << req.magics;
     } else {
         req.success = false;
     }
 }
 
+void nibaserver::server_processor::process(nibashared::message_learnmagic &req) {
+    req.magic = nibashared::staticdata::get().magic(req.static_id);
+    if (req.magic.description.empty()) {
+        req.magic.description = req.magic.name;
+    }
+    // by default we add it to the back
+    auto result = db_.create_magic(session_.player.value().name, req.magic, yield_);
+    if (!(req.success = result)) {
```

Comment:
tend to cause confusion    a few other places

Reply:
yeah some people don't like writing if like this, but I think the `!(...)` kinda clarifies it. Feel free to change them